src/SyntaxParser/ImportedClassesParser.php

```diff
@@ -25,6 +28,53 @@ public function parseImportedClasses(ContextualToken $firstToken)
         return $importedClasses;
     }
 
+    public function parseImportedClassesFromTokens(Tokens $tokens)
+    {
+        $importedClasses = new ImportedClasses();
+        $namespaceIndex = $tokens->getNextTokenOfKind(0, [[T_NAMESPACE]]);
+        $namespaceEndIndex = $tokens->getNextTokenOfKind($namespaceIndex, [';']);
+        $start = $tokens->getNextMeaningfulToken($namespaceIndex);
+        $end = $tokens->getPrevMeaningfulToken($namespaceEndIndex);
+        $namespace = $tokens->generatePartialCode($start, $end);
+        $importedClasses->setCurrentNamespace($namespace);
+        $tokensAnalyzer = new TokensAnalyzer($tokens);
+        $useIndices = $tokensAnalyzer->getImportUseIndexes();
+        foreach ($useIndices as $useIndex) {
+            $this->parseUseStatementFromTokens($useIndex, $tokens, $importedClasses);
+        }
+
+        return $importedClasses;
+    }
+
+    public function parseUseStatementFromTokens(int $start, Tokens $tokens, ImportedClasses $importedClasses)
+    {
+        $className = null;
+        $currentContent = '';
+        for ($i = $start + 1; $i < $tokens->count(); $i++) {
+            /** @var Token $token */
+            $token = $tokens[$i];
+            if ($token->getContent() === ';') {
+                break;
+            }
+            if ($token->isGivenKind(T_AS)) {
+                $className = $currentContent;
+                $currentContent = '';
+            } elseif (!$token->isWhitespace()) {
+                $currentContent .= $token->getContent();
+            }
+        }
```

Comment:
- this should also go to private method, `$something = null` declaration creates local state which indicates need for private method
- use full name instead of `$i`